browser-test/src/support/admin_programs.ts

```diff
@@ -724,23 +746,50 @@ export class AdminPrograms {
    * Creates a new program block as defined by {@link BlockSpec}.
    *
    * Prefer this method over {@link #addProgramBlock}.
+   *
+   * @param {string} programName Name of the program
+   * @param {BlockSpec} block Desired block settings
+   * @param {boolean} isProgramDisabled Defaults to false. Flag to determine if the program status is disabled or not
+   * @param {boolean} editBlockScreenDetails Defaults to true. If true the block name and description will be updated; if false they will not.
    */
   async addProgramBlockUsingSpec(
     programName: string,
     block: BlockSpec,
     isProgramDisabled: boolean = false,
+    editBlockScreenDetails: boolean = true,
```

Comment:
(non-blocking) This is kinda what I was worried about with the ProgramBlockSpec refactor... once you have more than one optional param, you have to specify all of them if you want to specify the last one, so you start hard-coding isProgramDisabled = false in places where it's not really helpful to the test. Which I guess isn't the worst thing, it's just more noise and doesn't scale too far.

I don't really have a suggestion, just thinking out loud. :)

Reply:
I'm starting thinking through ideas for a much broader change. I'm growing increasingly irritated at the run time of the tests. This is just a means to an end for now.